addons/workspace-dots/userscript.js

```diff
@@ -0,0 +1,84 @@
+export default async function ({ addon, console }) {
+  const blockly = await addon.tab.traps.getBlockly();
+
+  const oldFunction = blockly.getMainWorkspace().grid_.update;
+
+  function updateGrid() {
+    let themeSetting = addon.settings.get("theme");
+    let spacingSetting = addon.settings.get("useSpacing");
+    let spacingAmount = addon.settings.get("spacing");
+
+    // https://github.com/scratchfoundation/scratch-blocks/blob/develop/core/grid.js#L136
+    blockly.getMainWorkspace().grid_.update = function (scale) {
```

Comment:
why isnt this on `Blockly.Workspace.Grid.prototype.update`? Wouldn't that break when leaving the editor? At the very least it would probably break if you were to enable the addon after creating a custom block?

Reply:
Ohhh also I noticed you pollute everytime `updateGrid` is called. you should only pollute once and then call `workspace.grid.update()` when `updateGrid` is called.

Reply:
Remind me about this tmr

Reply:
This is hell. Read from https://discord.com/channels/751206349614088204/755131743136383107/1252657263257059419